ngraph_creator/operations/src/Channel_Shuffle.cpp

```diff
@@ -0,0 +1,89 @@
+#include <Channel_Shuffle.hpp>
+#define LOG_TAG "Channel_Shuffle"
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+Channel_Shuffle::Channel_Shuffle(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+bool Channel_Shuffle::validate() {
+    // check output type
+    if (!checkOutputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkOutputOperandType(0, (int32_t)OperandType::TENSOR_QUANT8_ASYMM)) {
+        return false;
+    }
+
+    // Check all input types
+    if (!checkInputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkInputOperandType(0, (int32_t)OperandType::TENSOR_QUANT8_ASYMM)) {
+        return false;
+    }
+    if (!checkInputOperandType(1, (int32_t)OperandType::INT32)) {
+        return false;
+    }
+    if (!checkInputOperandType(2, (int32_t)OperandType::INT32)) {
+        return false;
+    }
+
+    // Check input rank
+    const int64_t inputRank = getInputOperandDimensions(0).size();
+    if (inputRank > 4 || inputRank <= 0) return false;
```

Comment:
Add ALOGE with proper error message?

Reply:
Updated the commit with proper error handling.